plugins/mollie/app/controllers/payments/mollie_ideal_controller.rb

```diff
@@ -0,0 +1,160 @@
+# Mollie payment page
+class Payments::MollieIdealController < ApplicationController
+  before_action -> { require_plugin_enabled FoodsoftMollie }
+  skip_before_action :authenticate, :only => [:check]
+  skip_before_action :verify_authenticity_token, :only => [:check]
+  before_action :accept_return_to, only: [:new]
+  before_action :ordergroup, only: [:new, :create, :result]
+  before_action :transaction, only: [:result]
+  before_action :configure_api_key
+
+  def new
+    @amount = (params[:amount] or [10, -@ordergroup.get_available_funds].max)
+    @amount = [params[:min], params[:amount]].max if params[:min]
+  end
+
+  def create
+    # store parameters so we can redirect to original form on problems
+    session[:mollie_params] = params.select { |k, _| %w[amount label title fixed min text].include?(k) }.to_h
+
+    amount = params[:amount].to_f
+    amount = [params[:min].to_f, amount].max if params[:min]
+
+    redirect_to new_payments_mollie_path(session[:mollie_params]), alert: t('.invalid_amount') and return if amount <= 0
+
+    ft_type = FinancialTransactionType.find_by_id(FoodsoftConfig[:mollie][:financial_transaction_type]) || FinancialTransactionType.first
+
+    @transaction = FinancialTransaction.create!(
+      amount: nil,
+      ordergroup: @ordergroup,
+      user: @current_user,
+      payment_plugin: 'mollie',
+      payment_amount: amount,
+      # @todo payment_currency
+      payment_state: 'created',
+      financial_transaction_type_id: ft_type.id,
+      note: t('.controller.transaction_note', method: nil)
+    )
+
+    payment = Mollie::Payment.create(
+      amount: {
+        value: format('%.2f', amount),
+        currency: 'EUR' # @todo payment_currency
```

Comment:
I guess this would be a big refactoring ... I'd open a issue and reference it in the according places. I could also make it call smth like `FoodsoftMollie.currency`, than later only one place needs to be changed

Reply:
What about making this a Mollie configuration option (with a default to `EUR`), explaining that this needs to match Foodsoft's currency?
Available currencies at: https://docs.mollie.com/payments/multicurrency

Reply:
I had to realise that mollies method endpoint will always return the fee amount in EUR. I think for now the most straight forward thing is to make 'EUR' a constraint for the fee feature (I'd guess that's the most common case). If we find a way to retrieve the actual fee calculated by mollie we can move on to that. (Still calculating the right fee to increase the amount for different currencies before paying would be difficult)